src/main/java/com/google/sps/servlets/UserDataServlet.java

```diff
@@ -32,26 +38,39 @@
   /** Add/update a User into Datastore */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // TODO: Add posts to emails.
 
     // Get the input from the form.
     String name = request.getParameter("name");
     String email = request.getParameter("email");
     String university = request.getParameter("university");
     String subscription = request.getParameter("email-notif");
-    boolean subscribe = true;
 
     // Check if user wants to unsubscribe.
     if (subscription.equals("unsubscribe")) {
-      subscribe = false;
-    }
 
-    Entity userEntity = new Entity("User", email);
-    userEntity.setProperty("name", name);
-    userEntity.setProperty("university", university);
-    userEntity.setProperty("subscribe", subscribe);
+      // Remove user if in database.
+      Key userKey = KeyFactory.createKey("User", email);
+      try {
+        datastore.get(userKey);
+        datastore.delete(userKey);
+      } catch (EntityNotFoundException e) {
+        System.out.println("ERROR: User wants to unsubscribe but was never subscribed in the first place.");
```

Comment:
For now, this is okay, but ideally you want to use a logger for such errors.

Reply:
Gotcha! Just added a logger!